### PR TITLE
Use PIDs instead of command lines to identify processes in the agent's cgroup

### PR DESCRIPTION
**Pull request by narrieta**
Using command lines to identify those processes was unreliable and hard to maintain. PIDs should address those issues.

### CODE REVIEW

File: azurelinuxagent/common/cgroupconfigurator.py [hunk 1]
```diff
@@ -225,29 +237,70 @@ def __impl():
...
+
+            unexpected = []
+            for process in agent_cgroup:
+                if process in (daemon, extension_handler):
+                    continue
+                # check if the process is a command started by the agent or a descendant of one of those commands
+                current = process
+                while current != 0 and current not in agent_commands:
+                    current = self._get_parent(current)
+                if current == 0:
```
Comment (pgombar) [inline azurelinuxagent/common/cgroupconfigurator.py:267]: Isn't `current` supposed to be a PID? Why compare against 0?

File: azurelinuxagent/common/cgroupconfigurator.py [hunk 2]
```diff
@@ -225,29 +237,70 @@ def __impl():
...
+            agent_commands.update(shellutil.get_running_commands())
+
+            unexpected = []
+            for process in agent_cgroup:
+                if process in (daemon, extension_handler):
+                    continue
+                # check if the process is a command started by the agent or a descendant of one of those commands
+                current = process
+                while current != 0 and current not in agent_commands:
+                    current = self._get_parent(current)
```
Comment (pgombar) [inline azurelinuxagent/common/cgroupconfigurator.py:266]: Why not add `current` but its parent?

File: tests/common/test_cgroupconfigurator.py [hunk 3]
```diff
@@ -17,16 +17,21 @@
...
 from azurelinuxagent.common.cgroup import CGroup
-from azurelinuxagent.common.cgroupconfigurator import CGroupConfigurator
+from azurelinuxagent.common.cgroupconfigurator import CGroupConfigurator, UnexpectedProcessesInCGroupException
 from azurelinuxagent.common.cgroupstelemetry import CGroupsTelemetry
 from azurelinuxagent.common.exception import CGroupsException
+from azurelinuxagent.common.utils import shellutil
 from tests.common.mock_cgroup_commands import mock_cgroup_commands
 from tests.tools import AgentTestCase, patch, mock_sleep
-
+from tests.utils.miscelaneous_tools import format_processes, wait_for
```
Comment (pgombar) [inline tests/common/test_cgroupconfigurator.py:34]: Typo in module name, miscelaneous > miscellaneous

File: azurelinuxagent/common/cgroupconfigurator.py (hunk 1 above)
Comment (narrieta) [reply azurelinuxagent/common/cgroupconfigurator.py:267]: 0 is the PID of the scheduler; all processes descent from it

File: azurelinuxagent/common/cgroupconfigurator.py (hunk 2 above)
Comment (narrieta) [reply azurelinuxagent/common/cgroupconfigurator.py:266]: sorry, not sure i understand the question

File: tests/common/test_cgroupconfigurator.py (hunk 3 above)
Comment (narrieta) [reply tests/common/test_cgroupconfigurator.py:34]: thanks, will fix it

File: azurelinuxagent/common/cgroupconfigurator.py (hunk 2 above)
Comment (pgombar) [reply azurelinuxagent/common/cgroupconfigurator.py:266]: Didn't realize you are climbing up the process tree until you reach either the agent or the scheduler (parent process of all). Makes sense now!

File: tests/common/test_cgroupconfigurator.py (hunk 3 above)
Comment (narrieta) [reply tests/common/test_cgroupconfigurator.py]: done

**Review by pgombar — APPROVED** [review]

**Pull request merged**